lib/router.js

```diff
@@ -62,13 +65,15 @@ Router.prototype.match = function r_match(method, url) {
   method = this._sanitizeMethod(method);
   var matches = [];
 
-  var self = this;
+  var _this = this;
   this.stack.forEach(function eachRoute(route) {
-    if (!(method === route.method || route.method === self.ALL_METHODS)) {
+    if (!(method === route.method || route.method === _this.ALL_METHODS)) {
       return;
     }
 
-    if (route.path.test(url)) {
+    var groups = _this._routeMatch(url, route);
+    if (groups) {
+      route.handler.__params = groups;
```

Comment:
The handler can be a function too, not just a middleware object, are we safe for that option too?